org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/indexing/SourceIndexer.java

```diff
@@ -156,18 +161,8 @@ public void accept(ISourceType[] sourceTypes, PackageBinding packageBinding, Acc
 	}
 
 	public void resolveDocument() {
-		if (this.dom != null && getUnit() instanceof org.eclipse.jdt.internal.core.CompilationUnit unit) {
-			String reducedDOM = reduceDOM(dom);
-			try {
-				ASTParser astParser = ASTParser.newParser(AST.getJLSLatest()); // we don't seek exact compilation the more tolerant the better here
-				astParser.setSource(unit); // configure projects and so on
-				astParser.setSource(reducedDOM.toCharArray()); // trimmed contents
-				astParser.setStatementsRecovery(true);
-				astParser.setResolveBindings(true);
-				this.dom = astParser.createAST(null);
-			} catch (Exception e) {
-				ILog.get().error(e.getMessage(), e);
-			}
+		if (usedDomBasedIndexing() && this.dom != null && getUnit() instanceof org.eclipse.jdt.internal.core.CompilationUnit unit) {
```

Comment:
`useDomBasedIndexing()` is useless here as `this.dom` is null in case `usedDomBasedIndexing` is false.

Reply:
I understand that but it makes clear to the reader what we're actually looking for. Otherwise the reader needs to go through the whole file to see who sets this.dom anywhere and under what circumstances.